apps/chat/src/components/Chat/Publish/ApproveRequiredSection.tsx

```diff
@@ -0,0 +1,178 @@
+import { IconClipboard } from '@tabler/icons-react';
+import { useEffect, useState } from 'react';
+
+import { useTranslation } from 'next-i18next';
+
+import classNames from 'classnames';
+
+import { getPublicationId } from '@/src/utils/app/publications';
+
+import { BackendResourceType } from '@/src/types/common';
+import { FolderSectionProps } from '@/src/types/folder';
+import { Publication, PublicationInfo } from '@/src/types/publication';
+import { Translation } from '@/src/types/translation';
+
+import {
+  ConversationsActions,
+  ConversationsSelectors,
+} from '@/src/store/conversations/conversations.reducers';
+import { useAppDispatch, useAppSelector } from '@/src/store/hooks';
+import {
+  PublicationActions,
+  PublicationSelectors,
+} from '@/src/store/publication/publication.reducers';
+
+import CaretIconComponent from '../../Common/CaretIconComponent';
+import CollapsibleSection from '../../Common/CollapsibleSection';
+import {
+  ConversationPublicationResources,
+  PromptPublicationResources,
+} from './PublicationResources';
+
+import { some } from 'lodash-es';
+
+interface PublicationProps {
+  publication: PublicationInfo & Partial<Publication>;
+  resourceType: BackendResourceType;
+}
+
+const PublicationItem = ({ publication, resourceType }: PublicationProps) => {
+  const dispatch = useAppDispatch();
+
+  const selectedPublication = useAppSelector(
+    PublicationSelectors.selectSelectedPublication,
+  );
+  const selectedConversationIds = useAppSelector(
+    ConversationsSelectors.selectSelectedConversationsIds,
+  );
+
+  const [isOpen, setIsOpen] = useState(false);
+
+  const ResourcesComponent =
+    resourceType === BackendResourceType.CONVERSATION
+      ? ConversationPublicationResources
+      : PromptPublicationResources;
+
+  return (
+    <div className="flex flex-col gap-1">
+      <div
+        onClick={() => {
+          setIsOpen(true);
+          if (!isOpen) {
+            dispatch(
+              PublicationActions.uploadPublication({ url: publication.url }),
```

Comment:
I don't think that we need to upload publication 
see property `status` in interface `Entity`

Reply:
I only upload a publication if the publication itself has been closed. I don't go after it more than once

Reply:
@Alexander-Kezik ,
Ohh, why I can't close opened publication? It's bug

Reply:
will be fixed here
https://github.com/epam/ai-dial-chat/issues/1430